main.go

```diff
@@ -0,0 +1,28 @@
+package main
```

Comment:
> But how I originally thought about these was that they'd be separate processes, and the server would only accept registrations from the privileged places (ie. only registrations from localhost)

I would like to count @Liamsi in, since we all agree on this. 
> This also just seems like a configuration in the server, no?

I don't think so, since the server also has to accept lookup requests, too. I just have only 2 solutions for this:

1) We can check if the registration requests are originated from the localhost or not. (something like `conn.RemoteAddr() == conn.LocalAddr()`). But I don't know whether there is any type of attacks that can fake the remote address? 

2) Sign the request with registration bot's private key, and the key server verifies the request using the bot's public key. This way, the bot and the key server can run on 2 separated machines. 

What do you think?